In [55]:
import pandas as pd

In [56]:
main_football_data = pd.read_csv("MAIN_football.csv", sep=';', encoding='ISO-8859-1')

In [57]:
main_football_data['men_rank'] = main_football_data['men_rank'].astype(str).str.replace(',', '.').astype(float, errors='ignore')
main_football_data['women_rank'] = main_football_data['women_rank'].astype(str).str.replace(',', '.').astype(float, errors='ignore')

main_football_data = main_football_data[main_football_data['women_rank'] > 0]
main_football_data = main_football_data[main_football_data['men_rank'] > 0]

In [58]:
main_football_data

country  iso unicef_reg   unicef_gen_reg  un_reg       un_subreg  \
7     Afghanistan  AFG         SA  SA + SSA + MENA    Asia   Southern Asia   
8     Afghanistan  AFG         SA  SA + SSA + MENA    Asia   Southern Asia   
9     Afghanistan  AFG         SA  SA + SSA + MENA    Asia   Southern Asia   
10    Afghanistan  AFG         SA  SA + SSA + MENA    Asia   Southern Asia   
11    Afghanistan  AFG         SA  SA + SSA + MENA    Asia   Southern Asia   
...           ...  ...        ...              ...     ...             ...   
3134     Zimbabwe  ZWE        SSA  SA + SSA + MENA  Africa  Eastern Africa   
3135     Zimbabwe  ZWE        SSA  SA + SSA + MENA  Africa  Eastern Africa   
3136     Zimbabwe  ZWE        SSA  SA + SSA + MENA  Africa  Eastern Africa   
3137     Zimbabwe  ZWE        SSA  SA + SSA + MENA  Africa  Eastern Africa   
3138     Zimbabwe  ZWE        SSA  SA + SSA + MENA  Africa  Eastern Africa   

       inc_group inc_gen_group capital          lat  ... timezone  year  \
7     Low Income    Low Income   Kabul   34,5260109  ...  4:30:00  2012   
8     Low Income    Low Income   Kabul   34,5260109  ...  4:30:00  2013   
9     Low Income    Low Income   Kabul   34,5260109  ...  4:30:00  2014   
10    Low Income    Low Income   Kabul   34,5260109  ...  4:30:00  2015   
11    Low Income    Low Income   Kabul   34,5260109  ...  4:30:00  2016   
...          ...           ...     ...          ...  ...      ...   ...   
3134  Low Income    Low Income  Harare  -17,8567035  ...  2:00:00  2017   
3135  Low Income    Low Income  Harare  -17,8567035  ...  2:00:00  2018   
3136  Low Income    Low Income  Harare  -17,8567035  ...  2:00:00  2019   
3137  Low Income    Low Income  Harare  -17,8567035  ...  2:00:00  2020   
3138  Low Income    Low Income  Harare  -17,8567035  ...  2:00:00  2021   

        gii       gdp    temp host_future host_now  host_past     men_rank  \
7     0,738  2317,687  13,196           0        0          0   104.166667   
8     0,728  2385,706  14,254           0        0          0   182.166667   
9     0,718  2516,011  13,429           0        0          0   207.833333   
10    0,706  2534,919  14,066           0        0          0   193.166667   
11    0,692  2400,615  14,528           0        0          0   196.833333   
...     ...       ...     ...         ...      ...        ...          ...   
3134  0,532  2382,489  22,083           0        0          0   320.166667   
3135  0,535  2521,559  22,448           0        0          0   754.333333   
3136  0,533  2366,359   22,92           0        0          0  1177.875000   
3137  0,533  2254,493  22,384           0        0          0  1180.250000   
3138  0,532   2407,61   21,87           0        0          0  1160.501250   

      women_rank  
7        899.000  
8        899.000  
9        894.000  
10       889.000  
11       889.000  
...          ...  
3134    1213.750  
3135    1197.250  
3136    1171.500  
3137    1146.250  
3138    1132.225  

[2251 rows x 22 columns]

In [59]:
main_football_data.describe()

year  host_future     host_now    host_past     men_rank  \
count  2251.000000  2251.000000  2251.000000  2251.000000  2251.000000   
mean   2012.501555     0.012883     0.001777     0.133274   705.799265   
std       5.501787     0.112796     0.042126     0.339946   419.690380   
min    2003.000000     0.000000     0.000000     0.000000     5.333333   
25%    2008.000000     0.000000     0.000000     0.000000   380.083333   
50%    2013.000000     0.000000     0.000000     0.000000   631.583333   
75%    2017.000000     0.000000     0.000000     0.000000  1002.245834   
max    2021.000000     1.000000     1.000000     1.000000  1811.332500   

        women_rank  
count  2251.000000  
mean   1411.030687  
std     331.303000  
min     357.000000  
25%    1179.000000  
50%    1373.500000  
75%    1637.000000  
max    2225.000000

In [60]:
main_football_data['year'].value_counts().sort_values(ascending=False)

year
2021    147
2019    136
2020    135
2018    133
2015    126
2014    121
2007    121
2011    118
2016    118
2017    118
2012    118
2006    118
2013    117
2010    111
2005    107
2004    104
2008    104
2009    103
2003     96
Name: count, dtype: int64

# Наша база - 3139 записей о национальных футбольных сборных
# 22 столбца

# 154 уникальные страны

### Каждая запись содержит актуальные данные основных показателей определенной страны на определенный год
### страны встречаются от 1 до 19ти раз (19 раз встречаются 66 стран), остальные пропускают какие то года (не обязательно первые, потому что см. main_football_data['year'].value_counts())

### самый ранний встречаемый в данных год - 2003, самый поздний - 2021
# 

# на этапе очистки мы делим датасеты на мужской для дальнейшего анализа и женский для дальнейшего анализа

### после удаления нулей
### количество записей мужских сборных 3139 -> 3112 (-27 записей неиграющих сборных)
забавно, но в это время у половины из них играли женщины
### количество записей женских сборных 3129 -> 2263 (-876 записей неиграющих сборных)

# ну или можно снести сразу все, и анализировать только страны где играют оба пола

### тогда 3129 -> 2251

# ыыы столбцы

## Категориальные (номинальные): 
#### Страна, 
#### код Страны, 
#### код региона по классификации ЮНИСЕФ, (зачем?)
#### код региона по классификации ООН, (зачем?)
#### код региона по классификации ООН (зачем?)
#### столица (зачем?)
#### часовой пояс столицы (зачем?)


## Порядковые:
####  обобщенная классификация уровня дохода (всемирный банк), 
#### обобщенная классификация умного дохода (всемирный банк) 
#### host_past, host_now, host_future - дамми переменная на признание страны хозяйкой на ЧМ среди мужчин



## Числовые (количественные)
#### широта столицы, градусы (зачем?)
#### долгота столицы, градусы (зачем?)
#### координаты столицы, буквально список из широты и долготы (зачем?)
#### год 
#### индекс недерного неравенства GII
#### ввп на душу населения по ППС, доллары США (по текущему курсу) 
#### среднегодовая температура в столице за год (зачем?)
#### men_rank
#### women_rank
